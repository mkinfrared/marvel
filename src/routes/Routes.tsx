import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";

import { Pages } from "routes/Routes.type";
import { loadablePage } from "utils/loadable";

const Home = loadablePage(() => import("pages/Home"));
const Hero = loadablePage(() => import("pages/Hero"));
const NotFound = loadablePage(() => import("pages/NotFound"));

const Routes = () => {
  return (
    <Switch>
      <Route exact path={Pages.HOME} component={Home} />
      <Route path={`${Pages.HERO}:heroId`} component={Hero} />
      <Route path="/404" component={NotFound} />
      <Redirect to="/404" />
    </Switch>
  );
};

export default React.memo(Routes);
