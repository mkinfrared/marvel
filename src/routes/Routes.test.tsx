import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React from "react";
import { Router } from "react-router-dom";

import { Pages } from "routes/Routes.type";

import Routes from "./Routes";

jest.mock("pages/Home", () => () => "<Home />");

jest.mock("pages/Hero", () => () => "<Hero />");

jest.mock("pages/NotFound", () => () => "<NotFound />");

describe("<Routes />", () => {
  const history = createMemoryHistory();

  const Component = (
    <Router history={history}>
      <Routes />
    </Router>
  );

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should render Home page", async () => {
    const { findByText } = render(Component);
    const homePage = await findByText(/home/i);

    expect(homePage).toBeDefined();
  });

  it("should render Hero page", async () => {
    history.push(Pages.HERO + 42);

    const { findByText } = render(Component);
    const heroPage = await findByText(/hero/i);

    expect(heroPage).toBeDefined();
  });

  it("should render NotFound page", async () => {
    history.push("/404");

    const { findByText } = render(Component);
    const notFoundPage = await findByText(/notFound/i);

    expect(notFoundPage).toBeDefined();
  });
});
