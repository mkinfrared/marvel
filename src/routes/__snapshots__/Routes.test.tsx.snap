// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> should match the snapshot 1`] = `
<div>
  <div
    class="PageLoading"
  >
    <div
      class="Spinner"
      style="width: 50px; height: 50px;"
    />
  </div>
</div>
`;
